### PR TITLE
Update versions

### PR DESCRIPTION
**Pull request by xpepermint**
Updating versions and rebuilding manifests.

```json
{
  "alertmanager": "0.21.0",
  "blackboxExporter": "0.18.0",
  "grafana": "7.5.3",
  "kubeStateMetrics": "2.0.0",
  "nodeExporter": "1.1.2",
  "prometheus": "2.26.0",
  "prometheusAdapter": "0.8.4",
  "prometheusOperator": "0.47.0"
}
```

### CODE REVIEW

File: jsonnet/kube-prometheus/versions.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-  "nodeExporter": "1.1.1",
-  "prometheus": "2.25.0",
-  "prometheusAdapter": "0.8.3",
-  "prometheusOperator": "0.46.0"
+  "grafana": "7.5.3",
+  "kubeStateMetrics": "2.0.0",
+  "nodeExporter": "1.1.2",
+  "prometheus": "2.26.0",
+  "prometheusAdapter": "0.8.4",
+  "prometheusOperator": "0.47.0"
```
Comment (paulfantom) [inline jsonnet/kube-prometheus/versions.json:9]: This needs an upgrade in https://github.com/prometheus-operator/kube-prometheus/blob/main/jsonnet/kube-prometheus/jsonnetfile.json#L29 to `release-0.47`

File: jsonnet/kube-prometheus/versions.json [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
-  "grafana": "7.4.3",
-  "kubeStateMetrics": "2.0.0-rc.1",
-  "nodeExporter": "1.1.1",
-  "prometheus": "2.25.0",
-  "prometheusAdapter": "0.8.3",
-  "prometheusOperator": "0.46.0"
+  "grafana": "7.5.3",
+  "kubeStateMetrics": "2.0.0",
+  "nodeExporter": "1.1.2",
+  "prometheus": "2.26.0",
```
Comment (paulfantom) [inline jsonnet/kube-prometheus/versions.json:7]: This needs an upgrade in https://github.com/prometheus-operator/kube-prometheus/blob/main/jsonnet/kube-prometheus/jsonnetfile.json#L83

**Review by paulfantom — CHANGES_REQUESTED** [review]
To make it complete, you also need to modify https://github.com/prometheus-operator/kube-prometheus/blob/main/examples/thanos-sidecar.jsonnet#L10-L11 to bump thanos to 0.19 and do the same for thanos mixin in https://github.com/prometheus-operator/kube-prometheus/blob/main/jsonnet/kube-prometheus/jsonnetfile.json#L103

On top of that I think there is something wrong with jsonnet version used to generate files. Please remove `tmp/bin` directory and regenerate manifests.

**Comment by xpepermint** [comment]
Ups. I must say I haven't used `jsonnet` before so I'm sorry for that. I see some targets inside `jsonnetfile.json` point to a `master` branch. I left that part as it is, but I guess we wanna point that repos to a specific commit/version to be 100% safe.

**Comment by paulfantom** [comment]
> I guess we wanna point that repos to a specific commit/version to be 100% safe

We want that only for kube-prometheus `release-` branches, `main` should be as is.

---

It still looks that you are using an older jsonnet binary and not the one required to build manifests. Try running `make clean generate --always-make`.

File: manifests/kubernetes-prometheusRule.yaml [hunk 3]
```diff
@@ -510,26 +510,6 @@ spec:
...
-    - alert: KubeClientCertificateExpiration
-      annotations:
-        description: A client certificate used to authenticate to the apiserver is
-          expiring in less than 24.0 hours.
-        runbook_url: https://github.com/prometheus-operator/kube-prometheus/wiki/kubeclientcertificateexpiration
-        summary: Client certificate is about to expire.
-      expr: |
-        apiserver_client_certificate_expiration_seconds_count{job="apiserver"} > 0 and on(job) histogram_quantile(0.01, sum by (job, le) (rate(apiserver_client_certificate_expiration_seconds_bucket{job="apiserver"}[5m]))) < 86400
-      labels:
-        severity: critical
```
Comment (paulfantom) [inline manifests/kubernetes-prometheusRule.yaml]: This needs to be done in a separate PR and only for kops addon (by modifying https://github.com/prometheus-operator/kube-prometheus/blob/main/jsonnet/kube-prometheus/platforms/kops.libsonnet). We still want to have this for generic installation

File: manifests/kubernetes-prometheusRule.yaml [hunk 4]
```diff
@@ -719,31 +699,6 @@ spec:
...
-      annotations:
-        description: KubeControllerManager has disappeared from Prometheus target
-          discovery.
-        runbook_url: https://github.com/prometheus-operator/kube-prometheus/wiki/kubecontrollermanagerdown
-        summary: Target disappeared from Prometheus target discovery.
-      expr: |
-        absent(up{job="kube-controller-manager"} == 1)
-      for: 15m
-      labels:
-        severity: critical
```
Comment (paulfantom) [inline manifests/kubernetes-prometheusRule.yaml]: This **shouldn't** be removed from kube-prometheus at all. We already include [an addon](https://github.com/prometheus-operator/kube-prometheus/blob/main/jsonnet/kube-prometheus/platforms/kops.libsonnet#L17-L31) to enable gathering data from kcm and scheduler, the one lacking thing is documenting how to configure kops to allow kcm and scheduler to bind to `0.0.0.0` and not `127.0.0.1` so metrics can be gathered. That said documentation should be done in separate PR.

File: manifests/kubernetes-prometheusRule.yaml (hunk 3 above)
Comment (xpepermint) [reply manifests/kubernetes-prometheusRule.yaml]: How did that happen :). I have this change only in my `main` branch but not in the `update-versions`.

**Comment by xpepermint** [comment]
This was probably the worse PR I've ever created. It was supposed to be a 1min PR. Will do better next time, I promise :).

**Comment by paulfantom** [comment]
No worries :) I knew it out of top of my head only because I am working on automating the process of auto-updating components :)
